app/src/main/java/com/duckduckgo/app/browser/certificates/rootstore/TrustedCertificateStore.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2020 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.certificates.rootstore
+
+import android.net.http.SslCertificate
+import com.duckduckgo.app.browser.certificates.toX509Certificate
+import com.duckduckgo.app.browser.certificates.CertificateType
+import com.duckduckgo.app.browser.certificates.CertificateTypes
+import com.duckduckgo.app.browser.certificates.LetsEncryptCertificateProvider
+import timber.log.Timber
+import java.security.cert.*
+
+interface TrustedCertificateStore {
+    fun validateSslCertificateChain(sslCertificate: SslCertificate): CertificateValidationState
+}
+
+sealed class CertificateValidationState {
+    object IssuerExpired : CertificateValidationState()
+    object IssuerNotYetValid : CertificateValidationState()
+    object UntrustedChain : CertificateValidationState()
+    object TrustedChain : CertificateValidationState()
+}
+
+class TrustedCertificateStoreImpl(
+    private val letsEncryptCertificateProvider: LetsEncryptCertificateProvider
+) : TrustedCertificateStore {
+
+    /**
+     * Validates the given SSL certificate chain
+     * @param sslCertificate the SSL certificate to be validated
+     *
+     * @return [IssuerExpired] when any issuer in the [sslCertificate] chain is expired
+     * @return [IssuerNotYetValid] when any issuer in the [sslCertificate] chain is not yet valid
+     * @return [UntrustedChain] when we could not validate the [sslCertificate] certificate chain
+     * @return [TrustedChain] when SSL certificated chain is valiadted
+     */
+    override fun validateSslCertificateChain(sslCertificate: SslCertificate): CertificateValidationState {
+        return try {
+            validateSslCertificateChainInternal(sslCertificate)
+            CertificateValidationState.TrustedChain
+        } catch (t: Throwable) {
+            when (t) {
+                is CertificateExpiredException -> CertificateValidationState.IssuerExpired
+                is CertificateNotYetValidException -> CertificateValidationState.IssuerNotYetValid
+                else -> CertificateValidationState.UntrustedChain
+            }
+        }
+    }
+
+    private fun validateSslCertificateChainInternal(sslCertificate: SslCertificate) {
```

Comment:
Should this function also specify `@Throws` for consistency with the other function which declares it?